multinode-demo/remote_env.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+usage() {
+  echo -e "\\tUsage: source this-script <IP Address array> <username> [path to ssh keys]\\n"
+  echo -e "\\t <IP Address array>: A bash script that exports an array of IP addresses, SOLANA_NODE_IP_ADDRS. Elements of the array are public IP address of remote nodes."
+  echo -e "\\t <username>:         The username for logging into remote nodes."
+  echo -e "\\t [path to ssh keys]: The public/private key pair that remote nodes can use to perform rsync and ssh among themselves. Must contain pub, priv and authorized_keys.\\n"
+}
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+  echo -e "\\n\\tLooks like you are trying to run ${0}"
+  echo -e "\\tThe script must be sourced in your bash environment.\\n"
```

Comment:
Wait, why `source`?

Reply:
Oh I see.  Hmm, writing into my shell env is a little rude.  Can this be avoided?